mlir/include/mlir/Conversion/TosaToLinalg/TosaToLinalg.h

```diff
@@ -45,8 +48,12 @@ void registerTosaToLinalgPipelines();
 /// Populates conversion passes from TOSA dialect to Linalg dialect.
 void populateTosaToLinalgConversionPatterns(RewritePatternSet *patterns);
 
+enum class Conv2DKernelLayout { FHWC, HWCF };
+
 /// Populates conversion passes from TOSA dialect to Linalg named operations.
-void populateTosaToLinalgNamedConversionPatterns(RewritePatternSet *patterns);
+void populateTosaToLinalgNamedConversionPatterns(
```

Comment:
Could we make this take the `populate*` method take `TosaToLinalgNamedOptions` directly? Then we dont need the new `enum`. Also this is a entry point for *all* tosa to linalg named conversion patterns... So we want a stable ABI. Passing the options directly means that more optionality for controlling other named ops dont need API changes.